tensornetwork/matrixproductstates/base_mps.py

```diff
@@ -471,37 +471,61 @@ def apply_two_site_gate(self,
-              self.center_position, site1, site2))
-
-    gate_node = Node(gate, backend=self.backend)
-    node1 = Node(self.tensors[site1], backend=self.backend)
-    node2 = Node(self.tensors[site2], backend=self.backend)
-    node1[2] ^ node2[0]
-    gate_node[2] ^ node1[1]
-    gate_node[3] ^ node2[1]
-    left_edges = [node1[0], gate_node[0]]
-    right_edges = [gate_node[1], node2[2]]
-    result = node1 @ node2 @ gate_node
-    U, S, V, tw = split_node_full_svd(
-        result,
-        left_edges=left_edges,
-        right_edges=right_edges,
-        max_singular_values=max_singular_values,
-        max_truncation_err=max_truncation_err,
-        left_name=node1.name,
-        right_name=node2.name)
-    V.reorder_edges([S[1]] + right_edges)
-    left_edges = left_edges + [S[1]]
-    res = contract_between(U, S, name=U.name).reorder_edges(left_edges)
-    self.tensors[site1] = res.tensor
-    self.tensors[site2] = V.tensor
-    return tw
+    if (max_singular_values or max_truncation_err): 
+    
+        if (self.center_position not in (site1, site2)):
+            raise ValueError(
+              'center_position = {}, but gate is applied at sites {}, {}. '
+              'Truncation should only be done if the gate '
+              'is applied at the center position of the MPS'.format(
+                  self.center_position, site1, site2))
+            
+        else:
+            
+            gate_node = Node(gate, backend=self.backend)
+            node1 = Node(self.tensors[site1], backend=self.backend)
+            node2 = Node(self.tensors[site2], backend=self.backend)
+            node1[2] ^ node2[0]
+            gate_node[2] ^ node1[1]
+            gate_node[3] ^ node2[1]
+            left_edges = [node1[0], gate_node[0]]
+            right_edges = [gate_node[1], node2[2]]
+            result = node1 @ node2 @ gate_node
+            U, S, V, tw = split_node_full_svd(
+                result,
+                left_edges=left_edges,
+                right_edges=right_edges,
+                max_singular_values=max_singular_values,
+                max_truncation_err=max_truncation_err,
+                left_name=node1.name,
+                right_name=node2.name)
+            V.reorder_edges([S[1]] + right_edges)
+            left_edges = left_edges + [S[1]]
+            res = contract_between(U, S, name=U.name).reorder_edges(left_edges)
+            self.tensors[site1] = res.tensor
+            self.tensors[site2] = V.tensor
+            return tw
+            
+    else:
+        
+        gate_node = Node(gate, backend=self.backend)
+        node1 = Node(self.tensors[site1], backend=self.backend)
+        node2 = Node(self.tensors[site2], backend=self.backend)
+        node1[2] ^ node2[0]
+        gate_node[2] ^ node1[1]
+        gate_node[3] ^ node2[1]
+        left_edges = [node1[0], gate_node[0]]
+        right_edges = [gate_node[1], node2[2]]
+        result = node1 @ node2 @ gate_node
```

Comment:
This code is duplicated from above. Can you restructure the `if/else` statements to avoid the duplicate code?